sensing/pointcloud_preprocessor/test/test_distortion_corrector_node.cpp

```diff
@@ -0,0 +1,858 @@
+      rclcpp::spin_some(node_);
+      std::this_thread::sleep_for(std::chrono::milliseconds(10));
+    }
+  }
+
+  void TearDown() override {}
+
+  rclcpp::Time addMilliseconds(rclcpp::Time stamp, int ms)
+  {
+    auto ms_in_ns = rclcpp::Duration(0, ms * 1000000);
+    return stamp + ms_in_ns;
+  }
+
+  rclcpp::Time subtractMilliseconds(rclcpp::Time stamp, int ms)
+  {
+    auto ms_in_ns = rclcpp::Duration(0, ms * 1000000);
+    return stamp - ms_in_ns;
+  }
+
+  geometry_msgs::msg::TransformStamped generateTransformMsg(
+    const std::string & parent_frame, const std::string & child_frame, double x, double y, double z,
+    double qx, double qy, double qz, double qw)
+  {
+    geometry_msgs::msg::TransformStamped tf_msg;
+    tf_msg.header.stamp = node_->get_clock()->now();
+    tf_msg.header.frame_id = parent_frame;
+    tf_msg.child_frame_id = child_frame;
+    tf_msg.transform.translation.x = x;
+    tf_msg.transform.translation.y = y;
+    tf_msg.transform.translation.z = z;
+    tf_msg.transform.rotation.x = qx;
+    tf_msg.transform.rotation.y = qy;
+    tf_msg.transform.rotation.z = qz;
+    tf_msg.transform.rotation.w = qw;
+    return tf_msg;
+  }
+
+  std::vector<geometry_msgs::msg::TransformStamped> generateStaticTransformMsg()
+  {
+    return {
+      generateTransformMsg("base_link", "lidar_top", 5.0, 5.0, 5.0, 0.683, 0.5, 0.183, 0.499),
+      generateTransformMsg("base_link", "imu_link", 1.0, 1.0, 3.0, 0.278, 0.717, 0.441, 0.453)};
+  }
+
+  std::shared_ptr<geometry_msgs::msg::TwistWithCovarianceStamped> generateTwistMsg(
+    double linear_x, double angular_z, rclcpp::Time stamp)
+  {
+    auto twist_msg = std::make_shared<geometry_msgs::msg::TwistWithCovarianceStamped>();
+    twist_msg->header.stamp = stamp;
+    twist_msg->header.frame_id = "base_link";
+    twist_msg->twist.twist.linear.x = linear_x;
+    twist_msg->twist.twist.angular.z = angular_z;
+    return twist_msg;
+  }
+
+  std::shared_ptr<sensor_msgs::msg::Imu> generateImuMsg(
+    double angular_vel_x, double angular_vel_y, double angular_vel_z, rclcpp::Time stamp)
+  {
+    auto imu_msg = std::make_shared<sensor_msgs::msg::Imu>();
+    imu_msg->header.stamp = stamp;
+    imu_msg->header.frame_id = "imu_link";
+    imu_msg->angular_velocity.x = angular_vel_x;
+    imu_msg->angular_velocity.y = angular_vel_y;
+    imu_msg->angular_velocity.z = angular_vel_z;
+    return imu_msg;
+  }
+
+  std::vector<std::shared_ptr<geometry_msgs::msg::TwistWithCovarianceStamped>> generateTwistMsgs(
+    rclcpp::Time pointcloud_timestamp)
+  {
+    std::vector<std::shared_ptr<geometry_msgs::msg::TwistWithCovarianceStamped>> twist_msgs;
+    rclcpp::Time twist_stamp = subtractMilliseconds(pointcloud_timestamp, 5);
+
+    for (int i = 0; i < 6; ++i) {
+      auto twist_msg = generateTwistMsg(10.0 + i * 2, 0.02 + i * 0.01, twist_stamp);
+      twist_msgs.push_back(twist_msg);
+      // Make sure the twist stamp is not identical to any point stamp, and imu stamp
```

Comment:
I added the design as a note for users at https://github.com/autowarefoundation/autoware.universe/pull/7833/commits/a7c2a8b1907f2c276a33be248445806bdcda4ebe because I think creating additional function. 

Adding additional checks is a bit cumbersome as sometimes, there are cases with only imu, only twist, and empty pointcloud. I can work on that if you think it is a must  :)

Reply:
The case of identical pointstamps can be decided either way (to which imu/twist to match).
In this case is up to you which you decide, but if you declare something, keep it consistent (if you say you will do something, do it, it you want to leave something as TODO, say that, if something is not covered and will not, state that)

Reply:
Thanks, then I wouldn't add the additional check.

Reply:
@knzo25 
Is this one resolved?